src/types/composer.ts

```diff
@@ -310,10 +419,10 @@ export default class AlgokitComposer {
     return txnWithSigners
   }
 
-  private commonTxnBuildStep(params: CommonTxnParams, txn: algosdk.Transaction, suggestedParams: algosdk.SuggestedParams) {
-    if (params.lease) new Uint8Array(Buffer.from(params.lease))
+  private commonTxnBuildStep(params: CommonTransactionParams, txn: algosdk.Transaction, suggestedParams: algosdk.SuggestedParams) {
+    if (params.lease) txn.addLease(encodeLease(params.lease)!)
```

Comment:
This wasn't setting the lease :P, also now using the nice encodeLease helper from utils